qa/L0_metrics/cpu_metrics_test.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/python
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import re
+import threading
+import time
+import unittest
+from collections import defaultdict
+
+import numpy as np
+import requests
+import tritonclient.http as httpclient
+
+CPU_UTILIZATION = "nv_cpu_utilization"
```

Comment:
I had a good discussion with Jacky a while back on a different PR about how we should name constants and where they should be stored. The rule for python seems to be to name CONSTANTS with all caps. However, we also use this convention when reading environment variables (e.g., `os.environ.get("TRITONSERVER_IPADDR")`). Jacky also had a preference to store variables like this within the class definition. Link to full [discussion.](https://github.com/triton-inference-server/server/pull/6830#discussion_r1473665680). Wanted to know your thoughts on it.

Reply:
I'm not super opinionated on it. Personally I'd prefer:
```python
TRITONSERVER_IPADDR = os.environ.get("TRITONSERVER_IPADDR")
CPU_UTILIZATION = "nv_cpu_utilization
```
over this for consistency purposes:
```
_tritonserver_ipaddr = os.environ.get("TRITONSERVER_IPADDR")
CPU_UTILIZATION = "nv_cpu_utilization
```

I view `tritonserver_ipaddr` as a constant if only set once for the life of the test - but I wouldn't fight for one way over the other.

For now, I followed the example set by other scripts for easier find/replace in the future if we change our mind to reduce the number of special cases.

Reply:
Setting things within the class is a good idea too - but in this case I took an example that was using a method defined outside of the class, so I kept it constant. You could argue to move the helper method within the class too - but I don't think it will move the needle much one way or the other

Reply:
(note that this test is heavily copied from `pinned_memory_metrics_test.py` so I tried to keep it as similar as possible. It should make further decisions to change things easier to apply in multiple places.

Reply:
Great explanation!